### PR TITLE
Renamed WebAuthnManager.Config to WebAuthnManager.Configuration

### PR DESCRIPTION
**Pull request by dimitribouniol**
Renamed `WebAuthnManager.Config` to `WebAuthnManager.Configuration` to better match the rest of the ecosystem.

### CODE REVIEW

**Comment by dimitribouniol** [comment]
> This would be a breaking change, but given the library is v0.x.x it doesn't matter so much.
> 
> Is it worthwhile adding in CI for catching breaking changes now?

@0xTim said it should be fine due to alpha status: https://swift-open-source.slack.com/archives/C06L689TJ3A/p1708352712434729?thread_ts=1708352246.345259&cid=C06L689TJ3A

There are a probably a few other areas of API improvement I'll be suggesting over the next few days — would the CI step make those harder? If not I agree putting something in place now would be a good call, if only to point them out.

**Comment by dimitribouniol** [comment]
@adam-fowler Filed #50 for adding the CI step so we don't forget.

**Comment by marius-se** [comment]
Happy with renaming `Config` to `Configuration`! But I'm not sure how we handle breaking API changes in an alpha version, I'd hand that over to @0xTim

**Comment by marius-se** [comment]
> There are a probably a few other areas of API improvement I'll be suggesting over the next few days

Talking about other breaking API changes: I absolutely hate the fact that `finishRegistration` is `async` just because of `confirmCredentialIDNotRegisteredYet`. I built this back then because of step 26 [in the registration spec](https://w3c.github.io/webauthn/#reg-ceremony-assess-trust). I'd love to force the user to perform the check, but is it worth it? What do you think @dimitribouniol?

**Comment by dimitribouniol** [comment]
> Talking about other breaking API changes: I absolutely hate the fact that `finishRegistration` is `async` just because of `confirmCredentialIDNotRegisteredYet`. I built this back then because of step 26 [in the registration spec](https://w3c.github.io/webauthn/#reg-ceremony-assess-trust). I'd love to force the user to perform the check, but is it worth it? What do you think @dimitribouniol?

I think it's worth keeping around as it calls out the verification the caller must make to ensure uniqueness to prevent a known (and trivial) attack surface. That said, I think we should a) document this, and b) move the main implementation which is not async to a clearly named "unsafe" (or pre-checked) variant that can be called in synchronous code when the credentialIDs have already been verified. Does this make sense?

**Comment by 0xTim** [comment]
> I absolutely hate the fact that `finishRegistration` is `async` just because of `confirmCredentialIDNotRegisteredYet`

I wouldn't be too concerned over things like this and it's better to make it async and not do any async work than trying to make it work the other way around. It also opens up the possibilities of using actors without breaking the API

File: Sources/WebAuthn/WebAuthnManager+Config.swift
Comment (0xTim) [inline Sources/WebAuthn/WebAuthnManager+Config.swift:1]: Can you update the file name too?

**Review by 0xTim — CHANGES_REQUESTED** [review]
One nit

File: Sources/WebAuthn/WebAuthnManager+Config.swift
Comment (dimitribouniol) [reply Sources/WebAuthn/WebAuthnManager+Config.swift:1]: Done!

**Review by 0xTim — APPROVED** [review]
Thanks!

**Comment by 0xTim** [comment]
@dimitribouniol Just some merge conflicts then we're good to go

**Comment by dimitribouniol** [comment]
Lemme rebase in just a moment!

**Pull request merged**